### PR TITLE
Adding Detekt + Gradle 5 migration

### PR DESCRIPTION
**Pull request by juanimoli**
## Useful links
- [Contributing](https://github.com/mercadolibre/...) has more information and tips for a great pull request. This will give you a nice introduction about how to contribute with Andes.
- [Wiki](https://github.com/mercadolibre/...) has all the info you need to start with Andes. Even I heard there's a video there...

## Motivation
Adding Detekt for static code analysis

## Description
For adding Detekt on its latest version, we had to migrate to Gradle 5 plus AGP 3.4.2. For now, we're disabling R8 because we need to test it.

## Checks
Are you sure you followed all those steps? In such case, let's say with me "I solemnly declare that ...":
- [ ] I have coded an example inside the Demo App,
- [ ] I have added proper tests,
- [x] I have modified the [Changelog](https://github.com/mercadolibre/...) file,
- [ ] I have updated GitHub wiki,
- [ ] I have the explicit approval of one or more members of the UX Team,
- [ ] I have updated the version in gradle.properties file.

## Change type
This is easy, let us know what this code is about:
- [ ] This code adds a new component
- [x] This code includes improvements to an existent component
- [ ] This code improves or modifies ONLY the Demo App.

### CODE REVIEW

**Comment by furyGorgoryAdmin** [comment]
### Informative Menu <h2>  ### 

### Use any of these tags in a pull request description or leave a comment ### 
| Tag | Description |
|---|---|
| **[canary]**| Creates a build that can only be deployed with canary strategy. | 
| **[build]** |  Creates version or build from a branch that has been tagged as "no releasable". | 
| **[name:version-name]** |Specify the name of the build. For example: [build] [name:1.2.3-name] or [canary] [name:1.2.3-name].|
| **[major]** | To automatically increase the first position of the version. For example: [build] [major] or [canary] [major].|
| **[minor]** | To automatically increase the second position of the version. For example: [build] [minor] or [canary] [minor]. |
| **[patch]** | To automatically increase the third position of the version. For example: [build] [patch] or [canary] [patch].|  

[For more information about ci-cd](http://docs.furycloud.io/?lang=esp&section=businessassurance)
[For more information about semver](https://semver.org/#summary)

File: config/detekt.yml [hunk 1]
```diff
@@ -0,0 +1,591 @@
+build:
+  maxIssues: 400
```
Comment (juanimoli) [inline config/detekt.yml]: @jiurchuk esto no esta bien, el minimo por defecto es 10, le puse 400 porque tira ~360. Si no aplica, se puede dar de baja en este file

File: demoapp/build.gradle [hunk 2]
```diff
@@ -94,4 +94,8 @@ task copyChangelog(type: Copy) {
     }
 }
 
-tasks.copyChangelog.execute()
\ No newline at end of file
+//tasks.withType(type: Assemble) {
```
Comment (juanimoli) [inline demoapp/build.gradle]: @jiurchuk `.excecute()` no existe mas en Gradle 5, hay que ver como migrar esto

File: gradle.properties [hunk 3]
```diff
@@ -24,6 +24,11 @@ org.gradle.daemon=true
 org.gradle.caching=true
 ## Max workers for every build. This number can be tweaked
 org.gradle.workers.max=8
+## Disabling R8. TODO: test this properly to enable it
+android.enableR8=false
```
Comment (juanimoli) [inline gradle.properties]: @jiurchuk esto no deberia estar, lo desactive para que no me rompa R8, pero deberian hacer pruebas en la app con R8 prendido y en el modulo con R8 prendido para asegurar que todo ande

**Comment by juanimoli** [comment]
@jiurchuk btw, tenes un par de links rotos en el PR template

**Comment by jiurchuk** [comment]
> @jiurchuk btw, tenes un par de links rotos en el PR template

Thx!
https://github.com/mercadolibre/fury_andesui-android/pull/21

**Comment by jiurchuk** [comment]
@barriosnahuel it works!

File: demoapp/build.gradle (hunk 2 above)
Comment (jiurchuk) [reply demoapp/build.gradle]: Done on https://github.com/mercadolibre/fury_andesui-android/pull/39

**Pull request closed without merging**